lib/scholar/cluster/k_means.ex

```diff
@@ -73,7 +73,7 @@ defmodule Scholar.Cluster.KMeans do
       """
     ],
     seed: [
-      type: :integer,
+      type: {:custom, Scholar.Options, :seed, []},
```

Comment:
I think we should fully replace `:seed` with just accepting valid `Nx.Random.key`s